public/app/features/query/components/QueryGroup.tsx

```diff
@@ -452,3 +384,134 @@ const getStyles = stylesFactory(() => {
+  onOpenQueryInspector,
+}: QueryGroupTopSectionProps) {
+  const styles = getStyles();
+  const [isHelpOpen, setIsHelpOpen] = useState(false);
+  return (
+    <>
+      <div>
+        <div className={styles.dataSourceRow}>
+          <InlineFormLabel htmlFor="data-source-picker" width={'auto'}>
+            Data source
+          </InlineFormLabel>
+          <div className={styles.dataSourceRowItem}>
+            <DataSourcePickerWithPrompt
+              options={options}
+              onChange={async (ds, defaultQueries) => {
+                return await onDataSourceChange?.(ds, defaultQueries);
+              }}
+              isDataSourceModalOpen={Boolean(locationService.getSearchObject().firstPanel)}
+            />
+          </div>
+          {dataSource && (
+            <>
+              <div className={styles.dataSourceRowItem}>
+                <Button
+                  variant="secondary"
+                  icon="question-circle"
+                  title="Open data source help"
+                  onClick={() => setIsHelpOpen(true)}
+                  data-testid="query-tab-help-button"
+                />
+              </div>
+              <div className={styles.dataSourceRowItemOptions}>
+                <QueryGroupOptionsEditor
+                  options={options}
+                  dataSource={dataSource}
+                  data={data}
+                  onChange={(opts) => {
+                    onOptionsChange?.(opts);
+                  }}
+                />
+              </div>
+              {onOpenQueryInspector && (
+                <div className={styles.dataSourceRowItem}>
+                  <Button
+                    variant="secondary"
+                    onClick={onOpenQueryInspector}
+                    aria-label={selectors.components.QueryTab.queryInspectorButton}
+                  >
+                    Query inspector
+                  </Button>
+                </div>
+              )}
+            </>
+          )}
+        </div>
+        {dataSource && isAngularDatasourcePluginAndNotHidden(dataSource.uid) && (
+          <AngularDeprecationPluginNotice
+            pluginId={dataSource.type}
+            pluginType={PluginType.datasource}
+            angularSupportEnabled={config?.angularSupportEnabled}
+            showPluginDetailsLink={true}
+            interactionElementId="datasource-query"
+          />
+        )}
+      </div>
+      {isHelpOpen && (
+        <Modal title="Data source help" isOpen={true} onDismiss={() => setIsHelpOpen(false)}>
+          <PluginHelp pluginId={dsSettings.meta.id} />
+        </Modal>
+      )}
+    </>
+  );
+}
+
+interface DataSourcePickerWithPromptProps {
+  isDataSourceModalOpen?: boolean;
+  options: QueryGroupOptions;
+  onChange: (ds: DataSourceInstanceSettings, defaultQueries?: DataQuery[] | GrafanaQuery[]) => Promise<void>;
+}
+
+function DataSourcePickerWithPrompt({ options, onChange, ...otherProps }: DataSourcePickerWithPromptProps) {
```

Comment:
I thought about it, but since this component ain't exported I decided to leave it as is. Would probably extract it to a separate file if we needed it somewhere else.